src/partitions.jl

```diff
@@ -1,8 +1,396 @@
+next(p::FixedSetPartitions, s) = nextfixedsetpartition(p.s,p.m, s...)
+
+function nextfixedsetpartition(s::AbstractVector, m, a, b, n)
+    function makeparts(s, a)
+        part = [ similar(s,0) for k = 1:m ]
+        for i = 1:n
+            push!(part[a[i]], s[i])
+        end
+        return part
+    end
+
+    part = makeparts(s,a)
+
+    if m == 1
+        a[1] = 2
+        return (part, (a, b, n))
+    end
+
+    if a[end] != m
+        a[end] += 1
+    else
+        local j, k
+        for j = n-1:-1:1
+            if a[j]<m && b[a[j]+1]<j
+                break
+            end
+        end
+        if j>1
+            a[j]+=1
+            for p=j+1:n
+                if b[a[p]]!=p
+                    a[p]=1
+                end
+            end
+        else
+            for k=m:-1:2
+                if b[k-1]<b[k]-1
+                    break
+                end
+            end
+            b[k]=b[k]-1
+            b[k+1:m]=n-m+k+1:n
+            a[1:n]=1
+            a[b]=1:m
+        end
+    end
+
+    return (part, (a,b,n))
+end
+
+function nfixedsetpartitions(n::Int,m::Int)
+    numpart=0
+    for k=0:m
+        numpart+=(-1)^(m-k)*binomial(m,k)*(k^n)
+    end
+    numpart=div(numpart,factorial(m))
+    return numpart
+end
+
+#This function is still defined in Base because it is being used by Base.DSP
+#"""
+#Next integer not less than `n` that can be written as $\prod k_i^{p_i}$ for integers $p_1$, $p_2$, etc.
+#
+#For a list of integers i1, i2, i3, find the smallest
+#    i1^n1 * i2^n2 * i3^n3 >= x
+#for integer n1, n2, n3
+#"""
+#function nextprod(a::Vector{Int}, x)
+#    if x > typemax(Int)
+#        throw(ArgumentError("unsafe for x > typemax(Int), got $x"))
+#    end
+#    k = length(a)
+#    v = ones(Int, k)                  # current value of each counter
+#    mx = [nextpow(ai,x) for ai in a]  # maximum value of each counter
+#    v[1] = mx[1]                      # start at first case that is >= x
+#    p::widen(Int) = mx[1]             # initial value of product in this case
+#    best = p
+#    icarry = 1
+#
+#    while v[end] < mx[end]
+#        if p >= x
+#            best = p < best ? p : best  # keep the best found yet
+#            carrytest = true
+#            while carrytest
+#                p = div(p, v[icarry])
+#                v[icarry] = 1
+#                icarry += 1
+#                p *= a[icarry]
+#                v[icarry] *= a[icarry]
+#                carrytest = v[icarry] > mx[icarry] && icarry < k
+#            end
+#            if p < x
+#                icarry = 1
+#            end
+#        else
+#            while p < x
+#                p *= a[1]
+#                v[1] *= a[1]
+#            end
+#        end
+#    end
+#    best = mx[end] < best ? mx[end] : best
+#    return Int(best)  # could overflow, but best to have predictable return type
+#end
+
+"""
+Previous integer not greater than `n` that can be written as $\prod k_i^{p_i}$ for integers $p_1$, $p_2$, etc.
```

Comment:
You need to use doc"..." to be able to write latex like this.  That's the current travis failure.